Reply/app/src/main/java/com/example/reply/ui/theme/Theme.kt

```diff
@@ -15,95 +15,332 @@
+    background = backgroundDarkHighContrast,
+    onBackground = onBackgroundDarkHighContrast,
+    surface = surfaceDarkHighContrast,
+    onSurface = onSurfaceDarkHighContrast,
+    surfaceVariant = surfaceVariantDarkHighContrast,
+    onSurfaceVariant = onSurfaceVariantDarkHighContrast,
+    outline = outlineDarkHighContrast,
+    outlineVariant = outlineVariantDarkHighContrast,
+    scrim = scrimDarkHighContrast,
+    inverseSurface = inverseSurfaceDarkHighContrast,
+    inverseOnSurface = inverseOnSurfaceDarkHighContrast,
+    inversePrimary = inversePrimaryDarkHighContrast,
+    surfaceDim = surfaceDimDarkHighContrast,
+    surfaceBright = surfaceBrightDarkHighContrast,
+    surfaceContainerLowest = surfaceContainerLowestDarkHighContrast,
+    surfaceContainerLow = surfaceContainerLowDarkHighContrast,
+    surfaceContainer = surfaceContainerDarkHighContrast,
+    surfaceContainerHigh = surfaceContainerHighDarkHighContrast,
+    surfaceContainerHighest = surfaceContainerHighestDarkHighContrast,
+)
+
+fun isContrastAvailable(): Boolean {
+    return Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE
+}
+
+@Composable
+fun selectSchemeForContrast(isDark: Boolean,): ColorScheme {
+    val context = LocalContext.current
+    val uiModeManager = context.getSystemService(Context.UI_MODE_SERVICE) as UiModeManager
+    val contrastLevel = if (!isContrastAvailable()) 0.0 else uiModeManager.contrast
+
+    val colorScheme = when (contrastLevel as Float) {
+        in 0.0f..0.33f -> if (isDark)
+            darkScheme else lightScheme
+        in 0.34f..0.66f -> if (isDark)
+            mediumContrastDarkColorScheme else mediumContrastLightColorScheme
+        in 0.67f..1.0f -> if (isDark)
+            highContrastDarkColorScheme else highContrastLightColorScheme
+        else -> if (isDark) darkScheme else lightScheme
+    }
+    return colorScheme
+}
+@Composable
+fun ContrastAwareReplyTheme(
+    darkTheme: Boolean = isSystemInDarkTheme(),
+    // Dynamic color is available on Android 12+
+    dynamicColor: Boolean = false,
+    content: @Composable() () -> Unit
+) {
+    val replyColorScheme = when {
+        dynamicColor && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
+            val context = LocalContext.current
+            if (darkTheme) dynamicDarkColorScheme(context) else dynamicLightColorScheme(context)
+        }
+
+        else -> selectSchemeForContrast(darkTheme)
+    }
+    val view = LocalView.current
+    if (!view.isInEditMode) {
+        SideEffect {
+            val window = (view.context as Activity).window
+            window.statusBarColor = replyColorScheme.primary.toArgb()
+            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
+        }
+    }
+
+    MaterialTheme(
+        colorScheme = replyColorScheme,
+        typography = replyTypography, shapes = shapes,
+        content = content
+    )
+}
+
 @Composable
 fun ReplyTheme(
```

Comment:
Is the ReplyTheme still used anywhere?

Reply:
It was being used in previews but I've replaced all those references and removed ReplyTheme. Done.